### DISCUSSION TITLE
Resource filtering upon startup

### DISCUSSION THREAD

**Discussion by davidmirror-ops** [vmware-tanzu/octant#2611] [Ideas]
The ability to configure what resources are visible upon startup of the tool is the core of this discussion.

Is it anti-pattern for Octant to do that as it is indeed designed to give as much visibility as possible?

Is it a use case that could be addressed by the filter buttons Octant already has under each resource?

All comments, appreciations and ideas are welcome.

**Comment by vrabbi** [comment]
I think that it may be usefull to have a similar thing like in TMC where you can filter out/hide system objects. I think this can be helpfull in many cases
Hiding system resources would be like coredns,kube proxy,api server,etcd,kubernetes service etc.
This could clear up the ui when trying to debug just your apps running on the system and not be cluttered with the system objects
Having like a toggle switch to show/hide system resources in the preferences menu is how i would picture such a feature if it were to be implemented.

**Comment by davidberglund** [comment]
What I envisioned when bringing it up earlier was the ability to start up with a flag/config file and give it a list of resources that should be visible. Completely hiding other resources seems like it could be dangerous, so there should be something like a "clear filters" button like is common on e-commerce sites with search/filtering. 
The use case I see is both to be able to easily focus on the specific resources that you care about and when introducing k8s to people. For example if I use Contour I'm interested in HTTPProxy objects, but not ingress objects.

**Comment by davidmirror-ops** [comment]
Thank you @vrabbi and @davidberglund for your valuable input.

It would be great if @wwitzel3 adds his comments/thoughts on this, please :)

**Reply by wwitzel3** [reply]
I like this idea, I think it could be useful in many ways, the way I would see this being implemented as:

- establish a default location for the Octant preferences file
- ensure Octant reads preferences from that file if they are present
- provide a way to set the filter list of resources from the Preferences pane
- ensure the Preferences are saved on change to the preferences file
- create a visual indictor that clearly shows you are currently viewing a filtered set of resources
- allow easy toggle of the filter resource view
- allow easy opening the Preferences pane to edit the filtered list
- toggle the filter is also exposed in the Preferences pane

The idea is that you would be able to share the filtered view by sharing a minimal preference file that has the resources you want to show and setting the resource filter to enabled / true.
